rest/rest-sqs-testing-amazon-java-sdk/src/test/scala/org/elasticmq/rest/sqs/AmazonJavaSdkV2TestSuite.scala

```diff
@@ -32,16 +36,159 @@ class AmazonJavaSdkV2TestSuite extends SqsClientServerWithSdkV2Communication wit
+    clientV2.sendMessage(sendMessageRequest)
 
-    System.err.println(messages)
+    val message = receiveSingleMessageObject(queue.queueUrl(), requestedAttributes).orNull
+
+    // Then
+    message.body() shouldBe content
+    checkMessageAttributesMatchRequestedAttributes(messageAttributes, requestedAttributes, sendMessageRequest, message)
+  }
 
-    messages.messages().size() shouldBe 1
-    messages.messages().get(0).body() shouldBe "test msg 123"
+  private def receiveSingleMessageObject(queueUrl: String, requestedAttributes: List[String]): Option[Message] = {
+    clientV2
+      .receiveMessage(
+        ReceiveMessageRequest
+          .builder()
+          .queueUrl(queueUrl)
+          .messageAttributeNames(requestedAttributes.asJava)
+          .build()
+      )
+      .messages
+      .asScala
+      .headOption
+  }
+
+  private def checkMessageAttributesMatchRequestedAttributes(
+                                                              messageAttributes: Map[String, MessageAttribute],
+                                                              requestedAttributes: List[String],
+                                                              sendMessageRequest: SendMessageRequest,
+                                                              message: Message
+                                                            ) = {
+    val filteredSendMessageAttr =
+      filterBasedOnRequestedAttributes(requestedAttributes, sendMessageRequest.messageAttributes.asScala.toMap).asJava
+    val filteredMessageAttributes = filterBasedOnRequestedAttributes(requestedAttributes, messageAttributes)
+
+    message.messageAttributes should be(filteredSendMessageAttr)
+    message.messageAttributes.asScala.map { case (k, attr) =>
+      (
+        k,
+        if (attr.dataType.startsWith("String") && attr.stringValue != null) {
+          StringMessageAttribute(attr.stringValue).stringValue
+        } else if (attr.dataType.startsWith("Number") && attr.stringValue != null) {
+          NumberMessageAttribute(attr.stringValue).stringValue
+        } else {
+          BinaryMessageAttribute.fromByteBuffer(attr.binaryValue.asByteBuffer()).asBase64
+        }
+      )
+    } should be(filteredMessageAttributes.map { case (k, attr) =>
+      (
+        k,
+        attr match {
+          case s: StringMessageAttribute => s.stringValue
+          case n: NumberMessageAttribute => n.stringValue
+          case b: BinaryMessageAttribute => b.asBase64
+        }
+      )
+    })
+  }
+
+  private def filterBasedOnRequestedAttributes[T](
+                                                   requestedAttributes: List[String],
+                                                   messageAttributes: Map[String, T]
+                                                 ): Map[String, T] = {
+    if (requestedAttributes.contains("All")) {
+      messageAttributes
+    } else {
+      messageAttributes.view.filterKeys(k => requestedAttributes.contains(k)).toMap
```

Comment:
`filterKeys` is not available in scala 2.12, which we provide a build as well. Can you fix that?

Reply:
Changed from `messageAttributes.view.filterKeys` to `messageAtrributes.filterKeys`. It's already used on master in [AmazonJavaSdkTestSuite](https://github.com/softwaremill/elasticmq/blob/master/rest/rest-sqs-testing-amazon-java-sdk/src/test/scala/org/elasticmq/rest/sqs/AmazonJavaSdkTestSuite.scala#L391), so I suppose it should be ok for Scala 2.12?

Reply:
I'll check that. So far the build works good. Thanks for contribution!